internal/api/dto/custom_pricing_unit.go

```diff
@@ -0,0 +1,50 @@
+package dto
+
+import (
+	"github.com/shopspring/decimal"
+
+	"github.com/flexprice/flexprice/internal/types"
+)
+
+// CreateCustomPricingUnitRequest represents the request to create a new custom pricing unit
+type CreateCustomPricingUnitRequest struct {
+	Name           string           `json:"name" validate:"required"`
+	Code           string           `json:"code" validate:"required,len=3"`
+	Symbol         string           `json:"symbol" validate:"required,max=10"`
+	BaseCurrency   string           `json:"base_currency" validate:"required,len=3"`
+	ConversionRate *decimal.Decimal `json:"conversion_rate" validate:"required,gt=0"`
+	Precision      int              `json:"precision" validate:"gte=0,lte=8"`
+}
+
+// UpdateCustomPricingUnitRequest represents the request to update an existing custom pricing unit
+type UpdateCustomPricingUnitRequest struct {
+	Name      string       `json:"name,omitempty" validate:"omitempty"`
+	Symbol    string       `json:"symbol,omitempty" validate:"omitempty,max=10"`
+	Precision int          `json:"precision,omitempty" validate:"omitempty,gte=0,lte=8"`
+	Status    types.Status `json:"status" validate:"omitempty,oneof=published archived deleted"`
```

Comment:
can remove status
also we need to avoid oneof tag in the validate tags as it leads to issue in SDK generation. Hence we have implemented type specific validations and use them for such fields